File: java/Sixth/src/cooperation/TakeTrans.java
package cooperation;

public class TakeTrans {

    public static void main(String[] args) {

        Student studentJames = new Student("임채영", 5000);         // 학생 두명 생성
        Student studentTomas = new Student("정흥규", 10000);
        Student studentEdward = new Student("서환길", 30000);       // 택시 타는 과정을 추가로 구현하기 (나 혼자 코딩!)

        Bus bus100 = new Bus(100);                                  // 노선번호가 100번인 버스 생성
        studentJames.takeBus(bus100);                               // 임채영이 100번 버스를 탐
        studentJames.showInfo();                                    // 임채영의 정보 출력
        bus100.showInfo();                                          // 버스 정보 출력

        Subway subwayGreen = new Subway("2호선");                   // 노선번호가 2호선인 지하철 생성
        studentTomas.takeSubway(subwayGreen);                       // 정흥규가 2호선을 탐
        studentTomas.showInfo();                                    // 정흥규 정보 출력
        subwayGreen.showInfo();                                     // 지하철 정보 출력

        Taxi taxi4567 = new Taxi(4567);                             // 차량번호가 4567번인 택시 생성
        studentEdward.takeTaxi(taxi4567);                           // 서환길이 4567번 택시를 탐
        studentEdward.showInfo();                                   // 서환길 정보 출력
        taxi4567.showInfo();                                        // 택시 정보 출력
    }
}

File: python/python_oop_practice/p_oop_2/chap05/cooperation/take_trans.py
from class_module import *


def main():
    student_james = Student("James", 5000)
    student_tomas = Student("Tomas", 10000)

    bus100 = Bus(100)
    bus500 = Bus(500)

    subway_green = Subway(2)
    subway_blue = Subway(4)

    student_james.take_bus(bus100)
    student_tomas.take_subway(subway_green)

    student_james.show_info()
    student_tomas.show_info()

    bus100.show_bus_info()
    bus500.show_bus_info()

    subway_green.show_subway_info()

    student_edward = Student("Edward", 20000)
    well_taxi = Taxi("잘 간다 운수")
    student_edward.take_taxi(well_taxi)

    student_edward.show_info()
    well_taxi.show_taxi_info()


if __name__ == '__main__':
    main()
